beetsplug/web/__init__.py

```diff
@@ -37,7 +37,10 @@ def _rep(obj, expand=False):
     out = dict(obj)
 
     if isinstance(obj, beets.library.Item):
-        del out['path']
+        if app.config.get('INCLUDE_PATHS', False):
+            out['path'] = out['path'].decode('utf-8')
```

Comment:
You're right that everything needs to be converted to Unicode strings for JSON output. But, unfortunately, paths won't necessarily be encoded as UTF-8—on Unix, they can be arbitrary bytes. There are a few ways we might deal with this, such as base64-encoding the bytes in both directions. But one partial solution here would be to use `util.displayable_path`, which will at least avoid crashing here when the data contains un-decodeable bytes but will cause clients not to be able to find those paths.